src/my-bot.ts

```diff
@@ -1,38 +1,65 @@
+        let result = myShips.createBoard();
+        console.log(result);
+        return result;
 
-    public selectTarget(gamestate) {
-        var previousShot = gamestate.MyShots && gamestate.MyShots[gamestate.MyShots.length-1];
-        if(previousShot) {
-            return this.getNextTarget(previousShot.Position);
-        }
-        return { Row: "A", Column: 1 };  
+     
     }
 
-    private getNextTarget(position) {
-        var column = this.getNextColumn(position.Column);
-        var row = column === 1 ? this.getNextRow(position.Row) : position.Row;
-        return { Row: row, Column: column }
-    }
+    public selectTarget(gameState) {
+        try{
+            console.log("turn "+ (gameState.MyShots.length+1))
+            var mat:Matrix = new Matrix(gameState);
+            var ispreviousShot = gameState.MyShots && gameState.MyShots[gameState.MyShots.length-1];
+            if(ispreviousShot) {
+                var previousShot = gameState.MyShots[gameState.MyShots.length-1];
+                var previousShotAsCorrectClass = new Position(previousShot.Position.Row, previousShot.Position.Column);
+                if(previousShot.WasHit)
+                {
+                    if(!mat.isThereUnsunkShipAt(previousShotAsCorrectClass)) return this.getNextTarget(mat, gameState);
+                    else {
+                        // console.log(mat.board);
+                        return mat.returnUnsunkShot();
+                    }
+                }
+                else{
+                    for(let i = 1; i <= gameState.MyShots.length; i ++){
+                        if(gameState.MyShots[gameState.MyShots.length-i].WasHit){
+                            //this is the most recent hit;
+                            if(!mat.isThereUnsunkShipAt(new Position(gameState.MyShots[gameState.MyShots.length-i].Position.Row, gameState.MyShots[gameState.MyShots.length-i].Position.Column))) return this.getNextTarget(mat, gameState);
+                            else{
+                                // console.log(mat.board);
+                                return mat.returnUnsunkShot();
+                            }
+                        }
+                    }
+                }
+            }
+            return this.getNextTarget(mat, gameState);
 
-    private getNextRow(row) {
-        var newRow = row.charCodeAt(0) + 1;
-        if(newRow > 'J'.charCodeAt(0)) {
-            return 'A';
         }
-        return String.fromCharCode(newRow);
+        catch(err){
+            console.log(err);
+        }
     }
 
-    private getNextColumn(column) {
-        return column % 10 + 1;
+    private getNextTarget(mat:Matrix, gameState):Position {
+        mat.surroundHorizontalShips();
+        mat.surroundVerticalShips();
+        console.log(mat.board);
+        console.log(mat.ships);
+        //if(gameState.MyShots.length< 20)return TargetingMethods.targetMethodOpponentsMostRecentMiss(mat, gameState);
+        //else

```

Comment:
More upset about this commented out code, though.

(Why not have both of them live as an if/else, and have a `const isNewStrategy = true/false` to switch between them?